lib/node/runner/api/tx_operations.go

```diff
@@ -31,17 +33,51 @@ func (api NetworkHandlerAPI) GetOperationsByTxHashHandler(w http.ResponseWriter,
 	options := p.ListOptions()
 
 	if httputils.IsEventStream(r) {
-		event := fmt.Sprintf("txhash-%s", hash)
-		es := NewEventStream(w, r, renderEventStream, DefaultContentType)
-		ops, _ := api.getOperationsByTxHash(hash, options)
-		for _, op := range ops {
-			es.Render(op)
+		var blk *block.Block
+		es := NewEventStream(
+			w,
+			r,
+			func(args ...interface{}) ([]byte, error) {
+				if len(args) <= 1 {
+					return nil, fmt.Errorf("render: value is empty")
+				}
+				i := args[1]
+
+				if i == nil {
+					return nil, nil
+				}
+
+				if blk == nil {
+					if blk, err = api.getBlockByTxHash(hash); err != nil {
+						return nil, err
+					}
+				}
+
+				r := resource.NewOperation(i.(*block.BlockOperation))
+				r.Block = blk
+				return json.Marshal(r.Resource())
+			},
+			DefaultContentType,
+		)
+
+		var err error
+		if blk, err = api.getBlockByTxHash(hash); err == nil {
+			ops, _ := api.getOperationsByTxHash(hash, blk, options)
+			for _, op := range ops {
+				es.Render(op)
+			}
 		}
-		es.Run(observer.BlockOperationObserver, event)
+		es.Run(observer.BlockOperationObserver, fmt.Sprintf("txhash-%s", hash))
 		return
 	}
 
-	ops, cursor := api.getOperationsByTxHash(hash, options)
+	var blk *block.Block
+	if blk, err = api.getBlockByTxHash(hash); err != nil {
```

Comment:
How about that `blk, err = api.getBlockByTxHash` to line 35. 
EventStream and normal rendering can use it because of this tx hashes has same block.

Reply:
@anarcher The event stream should work if the block and operations is not saved yet.